projects/packages/my-jetpack/_inc/data/notices/use-notice.ts

```diff
@@ -1,7 +1,13 @@
 import { useContext, useEffect } from 'react';
 import { NoticeContext } from '../../context/notices/noticeContext';
 
-const useNotice = ( { message, options, isError } ) => {
+type NoticeType = {
+	message: string;
+	options: { status: string };
+	isError: boolean;
+};
+
+const useNotice: ( notice: NoticeType ) => void = ( { message, options, isError } ) => {
```

Comment:
The same above feedback applies to many other newly created functions in this PR.

Reply:
Thank you for such a detailed explanation of why this would be an improvement! I think Typescript took a backseat in my head due to the sheer amount of code that was being moved 😅, I will gladly improve this as I agree that it could definitely be better